zstd/zstd.go

```diff
@@ -1,33 +1,30 @@
-// +build cgo
-
+// Package zstd implements Zstandard compression.
 package zstd
 
 import (
 	"io"
 	"sync"
 
-	"github.com/DataDog/zstd"
+	zstdlib "github.com/klauspost/compress/zstd"
 	kafka "github.com/segmentio/kafka-go"
 )
 
 func init() {
 	kafka.RegisterCompressionCodec(NewCompressionCodec())
 }
 
-const (
-	Code = 4
+const Code = 4
 
-	DefaultCompressionLevel = zstd.DefaultCompression
-)
+const DefaultCompressionLevel = int(zstdlib.SpeedDefault)
```

Comment:
You cannot do this cast. The value is not interchangeable.

Reply:
I don't understand what you mean? 🤔

Reply:
Ah, now DefaultCompressionLevel is an integer, and before it was a plain constant, so this could be incompatible, right?

Reply:
Below you are doing `zstdlib.EncoderLevelFromZstd(level)` which assumes a reference zstd value.  

`zstdlib.SpeedDefault` has nothing to do with that. You can use `3` which is the reference library default value.

Reply:
😳 Now it makes even less sense.

```go
func EncoderLevelFromZstd(level int) EncoderLevel {
        switch {
        case level < 3:
                return SpeedFastest
        case level >= 3:
                return SpeedDefault
        }
        return SpeedDefault
}
```

Are you saying I can't use an int(SpeedDefault) as an argument to that function?

Reply:
Right, so EncoderLevel has its *own* scale for some reason. 🤨Thanks for mentioning that. 👍